### ISSUE TITLE
installation fail at seed Locations step

### ISSUE THREAD

**Issue by birutaibm** [opendatabio/opendatabio#187]
See the image:
![seedlocationsfail](https://user-images.githubusercontent.com/19666564/35623031-1802be0c-0671-11e8-86ae-4ed2e5e4b6ac.png)

I am running at Ubuntu 16.04 with the last version of mysql and supervisor. The installer give me warnings that mysql and supervisor need to be upgrade, but I ignore it. `supervisord --version` gives `3.2.0` and `mysql --version` gives `mysql  Ver 14.14 Distrib 5.7.21, for Linux (x86_64) using  EditLine wrapper`. The file /etc/supervisor/conf.d/opendatabio-worker.conf exists and have the content sugested by the intaller.

Another warning ignored (maybe related to the issue) is about max_allowed_packet and innodb_log_file_size, but the file /home/odbserver/.my.cnf contents is:
```
[mysqld]
max_allowed_packet = 100M
innodb_log_file_size = 300M

```

**Comment by andrechalom** [comment]
This error is very unexpected. This is certainly not related to supervisor or to the parameters you have listed. This *may* be related to the specific MySQL version you are using, I'll have to dig around for this. Can you check if this works for other Location insertions, such as using the web interface?

Also, as a general rule, please copy and paste the error messages as text. It works better than posting a screenshot, as it makes easier for me to copy it and paste in my terminal if I want to run the same code locally, or to copy and paste in google to try to find similar errors ;-)

**Comment by birutaibm** [comment]
I have no idea what is the problem! I try it using the web interface and give the same error:
SQLSTATE[22003]: Numeric value out of range: 1416 Cannot get geometry object from data you send to the GEOMETRY field (SQL: insert into `locations` (`name`, `altitude`, `datum`, `adm_level`, `notes`, `geom`, `lft`, `rgt`, `updated_at`, `created_at`) values (Biruta Country, 50, WGS84, 0, Meu país imaginário, GeomFromText('POLYGON((10 10,50 10,50 50,10 50,10 10),(20 20, 30 20, 30 30, 20 30,20 20))'), 1, 2, 2018-02-02 13:04:50, 2018-02-02 13:04:50))
There is no "numeric value" much big or small to be "out of the range"!
I will continue looking for something...

**Comment by andrechalom** [comment]
Can you log into the MySQL console and check the result of the following
command?

> DESCRIBE locations;

**Comment by birutaibm** [comment]
Ok, running in mysql console the error message changes:
```
mysql> insert into `locations` (`name`, `altitude`, `datum`, `adm_level`, `notes`, `geom`, `lft`, `rgt`, `updated_at`, `created_at`) values ('Biruta Country', 50, 'WGS84', 0, 'Meu país imaginário', ST_GeomFromText('POLYGON((10 10,50 10,50 50,10 50,10 10),(20 20, 30 20, 30 30, 20 30,20 20))'), 1, 2, '2018-02-02 13:04:50', '2018-02-02 13:04:50');
ERROR 1416 (22003): Cannot get geometry object from data you send to the GEOMETRY field
```
google this message, I found another github project with this problem discussed [here](https://github.com/sozialhelden/wheelmap/issues/355)
Please, check if it makes sense for you.

**Comment by andrechalom** [comment]
Yes, that seems to be related. I have edited the comment above, as the command I was intending to run was "DESCRIBE locations;". Can you check what is the data type of the "geom" column? Maybe you can try to issue "ALTER TABLE locations MODIFY COLUMN geom POLYGON" and see if this enables you to add the data? This will cause some trouble, as we need "geom" to be able to store POINTs, POLYGONs and MULTIPOLYGONs, but it will at least tell us where the cause of the problem is.

**Comment by birutaibm** [comment]
Ok, now it works... Let me show
1- what you ask me first:
```
mysql> describe locations;
+------------+--------------------+------+-----+---------+----------------+
| Field      | Type               | Null | Key | Default | Extra          |
+------------+--------------------+------+-----+---------+----------------+
| id         | int(10) unsigned   | NO   | PRI | NULL    | auto_increment |
| name       | varchar(191)       | NO   | MUL | NULL    |                |
| parent_id  | int(10) unsigned   | YES  |     | NULL    |                |
| lft        | int(10) unsigned   | NO   |     | 0       |                |
| rgt        | int(10) unsigned   | NO   |     | 0       |                |
| depth      | int(11)            | YES  |     | NULL    |                |
| altitude   | int(11)            | YES  |     | NULL    |                |
| adm_level  | int(11)            | NO   |     | NULL    |                |
| datum      | varchar(191)       | YES  |     | NULL    |                |
| uc_id      | int(10) unsigned   | YES  | MUL | NULL    |                |
| notes      | text               | YES  |     | NULL    |                |
| x          | decimal(8,2)       | YES  |     | NULL    |                |
| y          | decimal(8,2)       | YES  |     | NULL    |                |
| startx     | decimal(8,2)       | YES  |     | NULL    |                |
| starty     | decimal(8,2)       | YES  |     | NULL    |                |
| created_at | timestamp          | YES  |     | NULL    |                |
| updated_at | timestamp          | YES  |     | NULL    |                |
| geom       | geometrycollection | YES  |     | NULL    |                |
+------------+--------------------+------+-----+---------+----------------+
18 rows in set (0,18 sec)
```
2- Your suggestion:
```
mysql> ALTER TABLE locations MODIFY COLUMN geom POLYGON;
Query OK, 0 rows affected (1,05 sec)
Records: 0  Duplicates: 0  Warnings: 0

mysql> insert into `locations` (`name`, `altitude`, `datum`, `adm_level`, `notes`, `geom`, `lft`, `rgt`, `updated_at`, `created_at`) values ('Biruta Country', 50, 'WGS84', 0, 'Meu país imaginário', GeomFromText('POLYGON((10 10,50 10,50 50,10 50,10 10),(20 20, 30 20, 30 30, 20 30,20 20))'), 1, 2, '2018-02-02 13:04:50', '2018-02-02 13:04:50');
Query OK, 1 row affected, 1 warning (0,06 sec)
```
3- The new result of "describe":
```
mysql> describe locations;
+------------+------------------+------+-----+---------+----------------+
| Field      | Type             | Null | Key | Default | Extra          |
+------------+------------------+------+-----+---------+----------------+
| id         | int(10) unsigned | NO   | PRI | NULL    | auto_increment |
| name       | varchar(191)     | NO   | MUL | NULL    |                |
| parent_id  | int(10) unsigned | YES  |     | NULL    |                |
| lft        | int(10) unsigned | NO   |     | 0       |                |
| rgt        | int(10) unsigned | NO   |     | 0       |                |
| depth      | int(11)          | YES  |     | NULL    |                |
| altitude   | int(11)          | YES  |     | NULL    |                |
| adm_level  | int(11)          | NO   |     | NULL    |                |
| datum      | varchar(191)     | YES  |     | NULL    |                |
| uc_id      | int(10) unsigned | YES  | MUL | NULL    |                |
| notes      | text             | YES  |     | NULL    |                |
| x          | decimal(8,2)     | YES  |     | NULL    |                |
| y          | decimal(8,2)     | YES  |     | NULL    |                |
| startx     | decimal(8,2)     | YES  |     | NULL    |                |
| starty     | decimal(8,2)     | YES  |     | NULL    |                |
| created_at | timestamp        | YES  |     | NULL    |                |
| updated_at | timestamp        | YES  |     | NULL    |                |
| geom       | polygon          | YES  |     | NULL    |                |
+------------+------------------+------+-----+---------+----------------+
18 rows in set (0,06 sec)
```
4- And one more command to be sure that it works:
```
mysql> SELECT * from locations;
+----+----------------+-----------+-----+-----+-------+----------+-----------+-------+-------+-----------------------+------+------+--------+--------+---------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| id | name           | parent_id | lft | rgt | depth | altitude | adm_level | datum | uc_id | notes                 | x    | y    | startx | starty | created_at          | updated_at          | geom                                                                                                                                                                                  |
+----+----------------+-----------+-----+-----+-------+----------+-----------+-------+-------+-----------------------+------+------+--------+--------+---------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|  1 | Biruta Country |      NULL |   1 |   2 |  NULL |       50 |         0 | WGS84 |  NULL | Meu país imaginário   | NULL | NULL |   NULL |   NULL | 2018-02-02 13:04:50 | 2018-02-02 13:04:50 |                    $@      $@      I@      $@      I@      I@      $@      I@      $@      $@         4@      4@      >@      4@      >@      >@      4@      >@      4@      4@ |
+----+----------------+-----------+-----+-----+-------+----------+-----------+-------+-------+-----------------------+------+------+--------+--------+---------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0,00 sec)
```
Thank you.
Now we need to figure out How to be able to insert polygon or point at the same field of different registries. Maybe replace this field by two fields and select which one to use by php, forcing one of them to be null and use the another one? It solves the problem, but it seems no much elegant solution.

**Comment by andrechalom** [comment]
OK, the problem is that the entire idea behind using geometries was to
reduce data denormalization, and allow for a single database column to
store spatial data, be it a point or a multipolygon. And we need to cover
points for use as plots, and multipolygons for countries. Can you test this
behavior if you change the data type to "geometry" instead? ALTER TABLE
locations MODIFY COLUMN geom GEOMETRY;

**Comment by birutaibm** [comment]
It works, thank you. Follow I provide the commands used and their response for reference (the first command is only to clear the table):
```
mysql> delete from locations where id=1;
Query OK, 1 row affected (0,10 sec)

mysql> SELECT * from locations;
Empty set (0,00 sec)

mysql> describe locations;
+------------+------------------+------+-----+---------+----------------+
| Field      | Type             | Null | Key | Default | Extra          |
+------------+------------------+------+-----+---------+----------------+
| id         | int(10) unsigned | NO   | PRI | NULL    | auto_increment |
| name       | varchar(191)     | NO   | MUL | NULL    |                |
| parent_id  | int(10) unsigned | YES  |     | NULL    |                |
| lft        | int(10) unsigned | NO   |     | 0       |                |
| rgt        | int(10) unsigned | NO   |     | 0       |                |
| depth      | int(11)          | YES  |     | NULL    |                |
| altitude   | int(11)          | YES  |     | NULL    |                |
| adm_level  | int(11)          | NO   |     | NULL    |                |
| datum      | varchar(191)     | YES  |     | NULL    |                |
| uc_id      | int(10) unsigned | YES  | MUL | NULL    |                |
| notes      | text             | YES  |     | NULL    |                |
| x          | decimal(8,2)     | YES  |     | NULL    |                |
| y          | decimal(8,2)     | YES  |     | NULL    |                |
| startx     | decimal(8,2)     | YES  |     | NULL    |                |
| starty     | decimal(8,2)     | YES  |     | NULL    |                |
| created_at | timestamp        | YES  |     | NULL    |                |
| updated_at | timestamp        | YES  |     | NULL    |                |
| geom       | polygon          | YES  |     | NULL    |                |
+------------+------------------+------+-----+---------+----------------+
18 rows in set (0,01 sec)

mysql> ALTER TABLE locations MODIFY COLUMN geom GEOMETRY;
Query OK, 0 rows affected (1,08 sec)
Records: 0  Duplicates: 0  Warnings: 0

mysql> describe locations;
+------------+------------------+------+-----+---------+----------------+
| Field      | Type             | Null | Key | Default | Extra          |
+------------+------------------+------+-----+---------+----------------+
| id         | int(10) unsigned | NO   | PRI | NULL    | auto_increment |
| name       | varchar(191)     | NO   | MUL | NULL    |                |
| parent_id  | int(10) unsigned | YES  |     | NULL    |                |
| lft        | int(10) unsigned | NO   |     | 0       |                |
| rgt        | int(10) unsigned | NO   |     | 0       |                |
| depth      | int(11)          | YES  |     | NULL    |                |
| altitude   | int(11)          | YES  |     | NULL    |                |
| adm_level  | int(11)          | NO   |     | NULL    |                |
| datum      | varchar(191)     | YES  |     | NULL    |                |
| uc_id      | int(10) unsigned | YES  | MUL | NULL    |                |
| notes      | text             | YES  |     | NULL    |                |
| x          | decimal(8,2)     | YES  |     | NULL    |                |
| y          | decimal(8,2)     | YES  |     | NULL    |                |
| startx     | decimal(8,2)     | YES  |     | NULL    |                |
| starty     | decimal(8,2)     | YES  |     | NULL    |                |
| created_at | timestamp        | YES  |     | NULL    |                |
| updated_at | timestamp        | YES  |     | NULL    |                |
| geom       | geometry         | YES  |     | NULL    |                |
+------------+------------------+------+-----+---------+----------------+
18 rows in set (0,01 sec)

mysql> insert into `locations` (`name`, `altitude`, `datum`, `adm_level`, `notes`, `geom`, `lft`, `rgt`, `updated_at`, `created_at`) values ('Biruta Country', 50, 'WGS84', 0, 'Meu país imaginário', GeomFromText('POLYGON((10 10,50 10,50 50,10 50,10 10),(20 20, 30 20, 30 30, 20 30,20 20))'), 1, 2, '2018-02-02 13:04:50', '2018-02-02 13:04:50');
Query OK, 1 row affected, 1 warning (0,08 sec)

mysql> SELECT * from locations;
+----+----------------+-----------+-----+-----+-------+----------+-----------+-------+-------+-----------------------+------+------+--------+--------+---------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| id | name           | parent_id | lft | rgt | depth | altitude | adm_level | datum | uc_id | notes                 | x    | y    | startx | starty | created_at          | updated_at          | geom                                                                                                                                                                                  |
+----+----------------+-----------+-----+-----+-------+----------+-----------+-------+-------+-----------------------+------+------+--------+--------+---------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|  2 | Biruta Country |      NULL |   1 |   2 |  NULL |       50 |         0 | WGS84 |  NULL | Meu país imaginário   | NULL | NULL |   NULL |   NULL | 2018-02-02 13:04:50 | 2018-02-02 13:04:50 |                    $@      $@      I@      $@      I@      I@      $@      I@      $@      $@         4@      4@      >@      4@      >@      >@      4@      >@      4@      4@ |
+----+----------------+-----------+-----+-----+-------+----------+-----------+-------+-------+-----------------------+------+------+--------+--------+---------------------+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0,00 sec)
```
Please change the code that creates this table and close this issue. Thank you.

**Comment by andrechalom** [comment]
Fixed with a new migration file "2018_02_03_210145_change_datatype_of_locations_geom.php". Please note that this will not affect your current install, as all it does is change the data type from "geometrycollection" to "geometry". I will be submitting this along with some other changes in some hours.

**Comment by birutaibm** [comment]
I am reopen this issue due to a very strange behavior during install of current master branch. It seems that this error occurs again instead of the migration is correct applied. See below the end of installation output and the schema of the locations table, then let me know if you figure out what is happening.
```
Migrating: 2018_02_03_210145_change_datatype_of_locations_geom
Migrated:  2018_02_03_210145_change_datatype_of_locations_geom
Migrating: 2018_02_03_222339_create_world_location
Migrated:  2018_02_03_222339_create_world_location
Migrating: 2018_02_04_003042_add_parent_foreign_key_to_locations
Migrated:  2018_02_04_003042_add_parent_foreign_key_to_locations
Migrating: 2018_03_05_190132_add_affected_fields_to_user_jobs
Migrated:  2018_03_05_190132_add_affected_fields_to_user_jobs
Do you wish to import the default taxon and locations to the database? yes/[no] y
NOTICE: This will completely replace the current taxon and locations table. To proceed, type in uppercase PROCEED 
PROCEED
Downloading seed information...
Processing seed for taxons...
mysql: [Warning] Using a password on the command line interface can be insecure.
Processing seed for locations...
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1416 (22003) at line 58: Cannot get geometry object from data you send to the GEOMETRY field
Error processing database seed file!
Do you wish to seed the database with randomly generated test data? yes/[no] 
Changing storage area permissions...
********************************************

OpenDataBio has been successfully installed!

odbserver@rafael-note:~/opendatabio$ mysql -u opendatabio -p opendatabioEnter password: 
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 12
Server version: 5.7.21-0ubuntu0.16.04.1 (Ubuntu)

Copyright (c) 2000, 2018, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> describe locations;
+------------+--------------------+------+-----+---------+----------------+
| Field      | Type               | Null | Key | Default | Extra          |
+------------+--------------------+------+-----+---------+----------------+
| id         | int(10) unsigned   | NO   | PRI | NULL    | auto_increment |
| name       | varchar(191)       | NO   | MUL | NULL    |                |
| parent_id  | int(10) unsigned   | YES  |     | NULL    |                |
| lft        | int(10) unsigned   | NO   |     | 0       |                |
| rgt        | int(10) unsigned   | NO   |     | 0       |                |
| depth      | int(11)            | YES  |     | NULL    |                |
| altitude   | int(11)            | YES  |     | NULL    |                |
| adm_level  | int(11)            | NO   |     | NULL    |                |
| datum      | varchar(191)       | YES  |     | NULL    |                |
| uc_id      | int(10) unsigned   | YES  | MUL | NULL    |                |
| notes      | text               | YES  |     | NULL    |                |
| x          | decimal(8,2)       | YES  |     | NULL    |                |
| y          | decimal(8,2)       | YES  |     | NULL    |                |
| startx     | decimal(8,2)       | YES  |     | NULL    |                |
| starty     | decimal(8,2)       | YES  |     | NULL    |                |
| created_at | timestamp          | YES  |     | NULL    |                |
| updated_at | timestamp          | YES  |     | NULL    |                |
| geom       | geometrycollection | YES  |     | NULL    |                |
+------------+--------------------+------+-----+---------+----------------+
18 rows in set (0,01 sec)

```

**Comment by andrechalom** [comment]
This is definitely related to the different MySQL implementations / versions being used. I'm able to import the data with no problem in a MariaDB server, running either ArchLinux or Debian. And strangely, @betovicentini was able to import this same seed using Ubuntu 16.04, but I don't know what's his MySQL implementation or version.

**Comment by birutaibm** [comment]
I think the problem is more related with `opendatabio/data` so I comment there what I found.

**Comment by betovicentini** [comment]
I am using MariaDB latest version for Ubuntu. No problem importing seeds.

**Comment by andrechalom** [comment]
Should be fixed, please check.

**Comment by birutaibm** [comment]
Fixed by opendatabio/data#2
